docs/api/clojure/module.md

```diff
@@ -0,0 +1,235 @@
+;=> #'tutorial.module/out
+```
+
+
+By default, `context` is the CPU. If you need data parallelization, you can specify a GPU context or an array of GPU contexts like this `(m/module out {:contexts [(context/gpu)]})`
+
+Before you can compute with a module, you need to call `bind` to allocate the device memory and `init-params` or `set-params` to initialize the parameters. If you simply want to fit a module, you don’t need to call `bind` and `init-params` explicitly, because the `fit` function automatically calls them if they are needed.
+
+```clojure
+(let [mod (m/module out)]
+  (-> mod
+      (m/bind {:data-shapes (mx-io/provide-data train-data)
+               :label-shapes (mx-io/provide-label train-data)})
+      (m/init-params)))
+```
+
+Now you can compute with the module using functions like `forward`, `backward`, etc.
+
+## Training and Predicting
+
+Modules provide high-level APIs for training, predicting, and evaluating. To fit a module, call the `fit` function with some data iterators:
+
+```clojure
+(def mod (m/fit (m/module out) {:train-data train-data :eval-data test-data :num-epoch 1}))
+;; Epoch  0  Train- [accuracy 0.12521666]
+;; Epoch  0  Time cost- 8392
+;; Epoch  0  Validation-  [accuracy 0.2227]
+```
+
+You can pass in batch-end callbacks using batch-end-callback and epoch-end callbacks using epoch-end-callback in the `fit-params`. You can also set parameters using functions like in the fit-params like optimizer and eval-metric. To learn more about the fit-params, see the fit-param function options. To predict with a module, call `predict` with a DataIter:
+
+```clojure
+(def results (m/predict mod {:eval-data test-data}))
+(first results) ;=>#object[org.apache.mxnet.NDArray 0x3540b6d3 "org.apache.mxnet.NDArray@a48686ec"]
+
+(first (ndarray/->vec (first results))) ;=>0.08261358
+```
+
+The module collects and returns all of the prediction results. For more details about the format of the return values, see the documentation for the `predict` function.
+
+When prediction results might be too large to fit in memory, use the `predict-every-batch` API.
+
+```clojure
+(let [preds (m/predict-every-batch mod {:eval-data test-data})]
+  (mx-io/reduce-batches test-data
+                        (fn [i batch]
+                          (println (str "pred is " (first (get preds i))))
+                          (println (str "label is " (mx-io/batch-label batch)))
+                          ;;; do something
+                          (inc i))))
+```
+
+If you need to evaluate on a test set and don’t need the prediction output, call the `score` function with a data iterator and an eval metric:
+
+```clojure
+(m/score mod {:eval-data test-data :eval-metric (eval-metric/accuracy)}) ;=>["accuracy" 0.2227]
+```
+
+This runs predictions on each batch in the provided data iterator and computes the evaluation score using the provided eval metric. The evaluation results are stored in metric so that you can query later.
```

Comment:
... stored in `metric`, which you can query later.

how? or is that a known thing for folks using clojure?

Reply:
It's the eval-metric object ... I'll try to clarify